grc/upload/__init__.py

```diff
@@ -85,14 +98,25 @@ def uploadInfoPage(section_url: str):
 
             DataStore.save_application(application_data)
 
-            return local_redirect(url_for('upload.uploadInfoPage', section_url=section.url) + '#file-upload-section')
+            if has_password:
+                return local_redirect(url_for('upload.documentPassword', section_url=section.url))
+            else:
+                return local_redirect(url_for('upload.uploadInfoPage', section_url=section.url) + '#file-upload-section')
 
         elif form.button_clicked.data == 'Save and continue':
             if len(files) > 0:
                 return local_redirect(url_for('taskList.index'))
             else:
                 form.documents.errors.append('Select a JPG, BMP, PNG, TIF or PDF file smaller than 10MB')
 
+    else:
+
+        # Delete any password protected files that aren't unlocked
+        password_protected_files = [file for file in section.file_list(application_data.uploads_data) if file.password_required]
+        for file in password_protected_files:
+            application_data = delete_file(application_data, file.aws_file_name, section)
+        DataStore.save_application(application_data)
+
     return render_template(
         f"upload/{section.html_file}",
```

Comment:
There are two sections of code which are inconsistent.

This section deletes any password-protected files.
Another section (on `_upload-files-section.html`) puts the padlock symbol next to any password-protected files and asks the user to go to the page to enter the passwords.

Can we have one, or the other, but not both.

Let's talk to Tor and Spence/Alex to decide on the which workflow is clearest / most consistent.